### PR TITLE
Auto gradient accumulation

### LINKED ISSUE #371: Try using auto_find_batch_size
Feature of accelerate library but already built into HF training arguments

### PR DESCRIPTION
**Pull request by isaac091**
Adds option to find and use the highest batch size with the corresponding number of gradient accumulation steps. 

The new code mostly mirrors `accelerate` and the base Trainer's implementation of the `auto_find_batch_size` option, but adjusting the number of gradient accumulation steps needs to happen at the same time as adjusting the batch size as far as I can tell, so I wasn't able to just use the `auto_find_batch_size` option and adjust the grad. acc. steps after the fact.

Closes #371

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/silnlp/424)

### CODE REVIEW

**Review by ddaspit — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/silnlp/424#-:-O3Er2lR5nXxsd6f2d5M:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @isaac091)
